ch4/problem_2.py

```diff
@@ -1,2 +1,15 @@
-def unique():
-    pass
\ No newline at end of file
+def unique(iterable, key=None):
+    uniqueness = {1}
```

Comment:
`uniqueness = set()` would be simpler than creating with one entry then removing. Otherwise very clean solution.